R/PipeOpVtreat.R

```diff
@@ -0,0 +1,217 @@
+#' @title PipeOpVtreat
+#'
+#' @usage NULL
+#' @name mlr_pipeops_vtreat
+#' @format [`R6Class`] object inheriting from [`PipeOpTaskPreproc`]/[`PipeOp`].
+#'
+#' @description
+#' Provides an interface to the vtreat package.
+#'
+#' [`PipeOpVtreat`] naturally works for [classification Tasks][mlr3::TaskClassif] and [regression Tasks][mlr3::TaskRegr].
+#' Internallly [`PipeOpVtreat`] follows the fit/prepare interface of vtreat.
+#'
+#' @section Construction:
+#' ```
+#' PipeOpVreat$new(id = "vtreat", param_vals = list())
+#' ```
+#'
+#' * `id` :: `character(1)`\cr
+#'   Identifier of resulting object, default `"vtreat"`.
+#' * `param_vals` :: named `list`\cr
+#'   List of hyperparameter settings, overwriting the hyperparameter settings that would otherwise be set during construction. Default `list()`.
+#'
+#' @section Input and Output Channels:
+#' Input and output channels are inherited from [`PipeOpTaskPreproc`].
+#'
+#' The output is the input [`Task`][mlr3::Task] with all affected features "prepared" by vtreat.
+#' If vtreat found "no usable vars", the input [`Task`][mlr3::Task] is returned unaltered.
+#'
+#' @section State:
+#' The `$state` is a named `list` with the `$state` elements inherited from [`PipeOpTaskPreproc`], as well as:
+#' * `treatment_plan` :: object of class `vtreat_pipe_step` | `NULL`\cr
+#'   The treatment plan as constructed by vtreat based on the training data, i.e., an object of class `treatment_plan`.
+#'   If vtreat found "no usable vars" and designing the treatment would have failed, this is `NULL`.
+#'
+#' @section Parameters:
+#' The parameters are the parameters inherited from [`PipeOpTaskPreproc`], as well as:
+#' * `recommended` :: `logical(1)`\cr
+#'   Whether only the "recommended" constructed columns should be returned, i.e., non constant
+#'   variables with a significance value smaller than vtreat's threshold.
+#' * `cols_to_copy` :: `list`\cr
+#'   List of extra columns to copy.
+#' * `params` :: `list`\cr
```

Comment:
split this up into the actual paramset. If this turns out to be heavily dependent on the task type, split this class up into two (classif & regr, maybe also classif binary vs multiclass, but would be nice if we can avoid that), probably inheriting from a common base class.